service/dap/server.go

```diff
@@ -2157,10 +2157,17 @@ func (s *Session) onVariablesRequest(request *dap.VariablesRequest) {
 }
 
 func (s *Session) maybeLoadResliced(v *fullyQualifiedVariable, start, count int) (*fullyQualifiedVariable, error) {
-	if start == 0 && count == len(v.Children) {
-		// If we have already loaded the correct children,
-		// just return the variable.
-		return v, nil
+	if start == 0 {
+		want := count
+		if v.Kind == reflect.Map {
+			// For maps, we need to have 2*count children since each key-value pair is two variables.
+			want *= 2
```

Comment:
This does not handle the two-child-per-pair map any differently. It will also say that is has 60 indexed children, but when 0-60 are requested, 120 children will be returned (2 for each KV pair instead of 1).

Reply:
Ah, I see now. The client is told there will be 60 child (child=kv pair), but then the server returns 120 separate children, even though it is only asked for 60. Then yeah, the math here will be the same.